src/functions/members.js

```diff
@@ -0,0 +1,49 @@
+import {db} from '../Firebase/FirebaseConfig';
+import { collection, getDocs, doc, getDoc, setDoc, addDoc, deleteDoc } from 'firebase/firestore';
+
+async function doesMemberExist(snapshot) {
+    if (!snapshot.exists()) throw new Error('Member not found.');
```

Comment:
This is where Typescript can help you. Looking at this, we have no idea the structure of this data.

Reply:
This is probably better sorted as a utils for firebase. Also better off as a function that returns true or false like the name indicates instead of throwing an unexpected generic error. Other functions need to have their own specific error messages.